### PR TITLE
[NER] Change Validation loss to F1

### PR DESCRIPTION
**Pull request by cheungdaven**
*Description of changes:*

1. Change validation loss to F1 loss.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/optimization/utils.py [hunk 1]
```diff
@@ -263,6 +264,8 @@ def get_metric(
     metric_name = metric_name.lower()
     if metric_name in [ACC, ACCURACY, OVERALL_ACCURACY]:
         return torchmetrics.Accuracy(), None
+    elif metric_name == OVERALL_F1:
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/optimization/utils.py:267]: Change to `NER_TOKEN_F1`?

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 2]
```diff
@@ -1421,6 +1422,9 @@ def _top_k_average(
         strict_loading=True,
         standalone=True,
     ):
+        if self._problem_type is not None and self._problem_type == NER:
+            validation_metric_name = OVERALL_F1  # seqeval only support overall_f1
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/predictor.py:1426]: There is no need to add the additional "if" here. If the problem is NER, the proper validation_metric and eval_metric should be set here: https://github.com/autogluon/autogluon/blob/10b39aceb35e943f7c62f095f0005b5b42aa8aa4/multimodal/src/autogluon/multimodal/utils/metric.py#L104-L105

Comment (cheungdaven) [reply multimodal/src/autogluon/multimodal/predictor.py:1426]: > There is no need to add the additional "if" here. If the problem is NER, the proper validation_metric and eval_metric should be set here:
> 
> https://github.com/autogluon/autogluon/blob/10b39aceb35e943f7c62f095f0005b5b42aa8aa4/multimodal/src/autogluon/multimodal/utils/metric.py#L104-L105

in line 1475, best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name], the evaluate function is called and validation_metric_name is used. However, in the evaluate function, seqeval is used and only overall_f1, overall_accuracy, overall_precision, overall_recall, and tag name are valid evaluation metric. NER_TOKEN_F1 is not supported, that is why I used OVERALL_F1 as validation_metric_name and eval_metric_name. To separate them, we have to do this trick here.

**Review by sxjscience — APPROVED** [review]
LGTM

File: docs/tutorials/multimodal/text_prediction/ner.md [hunk 3]
```diff
@@ -45,6 +45,7 @@ model_path = f"./tmp/{uuid.uuid4().hex}-automm_ner"
 predictor = MultiModalPredictor(problem_type="ner", label=label_col, path=model_path)
 predictor.fit(
     train_data=train_data,
+    hyperparameters={'model.ner_text.checkpoint_name':'bert-base-cased'},
```
Comment (tonyhoo) [inline docs/tutorials/multimodal/text_prediction/ner.md:48]: Can we make this default? Not neccessary for 0.6.2 release, but better to refactor and make it simpler for users going forward (e.g. 0.7 release)

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 2 above)
Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/predictor.py:1428]: Updated that we added a `#FIXME` in the source code. So I'm resolving this and we will revisit later.

File: docs/tutorials/multimodal/text_prediction/ner.md (hunk 3 above)
Comment (cheungdaven) [reply docs/tutorials/multimodal/text_prediction/ner.md:48]: It uses deberta base by default. I set it to bert-base-cased as it gives better demonstration results given such a short training time (300s) just for tutorial purpose.

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**